helmclient.go

```diff
@@ -259,6 +296,28 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	} else if !installTiller && upgradeTiller {
```

Comment:
This is the new code for the upgrade.

Reply:
Just wondering if it would not be better to split this big function in small chunks (moving out the upgrade, install,... parts) and improve readability

Reply:
Good call. I'll take a look at splitting it.

Reply:
I've split out the new code to `installTiller` and `upgradeTiller` methods. More of the existing code could be split out but I think its better to do as separate PRs. I'll continue if I have time.